gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceExtractor.java

```diff
@@ -775,6 +790,122 @@ public boolean bulkApiLogin() throws Exception {
   }
 
   /**
+   * Get a buffered reader wrapping the query result stream for the result with the specified index
+   * @param index index the {@link #bulkResultIdList}
+   * @return a {@link BufferedReader}
+   * @throws AsyncApiException
+   */
+  private BufferedReader getBulkBufferedReader(int index) throws AsyncApiException {
+    return new BufferedReader(new InputStreamReader(
+        this.bulkConnection.getQueryResultStream(this.bulkJob.getId(), this.bulkResultIdList.get(index).getBatchId(),
+            this.bulkResultIdList.get(index).getResultId()), ConfigurationKeys.DEFAULT_CHARSET_ENCODING));
+  }
+
+  /**
+   * Fetch a batch of records
```

Comment:
It's better to make it clearer that this batch isn't the chunk batch, but the extractor is grouping the records